src/stream.ts

```diff
@@ -40,22 +41,29 @@ export class Streamer {
     const onConnected = () => {
       this.logDebugMessage('Stream connected')
       this.eventBus.emit(Event.CONNECTED)
+      this.reconnectAttempts = 0
     }
 
     const onDisconnect = () => {
       clearInterval(this.readTimeoutCheckerId)
       const reconnectDelayMs = getRandom(1000, 10000)
+      this.reconnectAttempts++
       this.logDebugMessage('Stream disconnected, will reconnect in ' + reconnectDelayMs + 'ms')
       if (!this.disconnectEventEmitted) {
         this.eventBus.emit(Event.DISCONNECTED)
         this.disconnectEventEmitted = true
       }
+      if (this.reconnectAttempts >= 5 && this.reconnectAttempts % 5 === 0) {
+        this.logErrorMessage(
+          `Reconnection failed after ${this.reconnectAttempts} attempts; attempting further reconnections.`
+        )
+      }
```

Comment:
Super minor, but a line break before and after an `if` helps avoid *Wall of Code™️* situations and makes the code a lot more visually skimmable.
```suggestion

      if (this.reconnectAttempts >= 5 && this.reconnectAttempts % 5 === 0) {
        this.logErrorMessage(
          `Reconnection failed after ${this.reconnectAttempts} attempts; attempting further reconnections.`
        )
      }

```